src/main/java/org/apache/commons/cli/converters/Verifier.java

```diff
@@ -0,0 +1,74 @@
+/*
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package org.apache.commons.cli.converters;
+
+import java.util.regex.Pattern;
+
+/**
+ * The definition of the functional interface to call when verifying a string
+ * for input Like {@code Predicate<String>} but can throw a RuntimeException.
+ * @since 1.7
+ */
+@FunctionalInterface
+public interface Verifier {
```

Comment:
Probably more flexible if this extends `Predicate<String>`.

Reply:
Note that the advantage of a custom interface here is that we are cementing the design to only test `String` input, as opposed `CharSequence` for example.